### ISSUE TITLE
Have a consistent version numbers for skygear-server, plugins and SDK

### ISSUE THREAD

**Issue by chpapa** [SkygearIO/features#198]
# Description

Currently it is difficult to know which versions of plugins will works with which version of skygear-server, problem happens all the time especially if you look at support emails.

I think we can have a consistent version number. One idea is major and minor number must match for skygear-server, plugins and client SDK to be compatible. (In the schema of MAJOR.MINOR.PATCH)

Here are several advantage:

1. It is easy for users to identify, 1.14.5 skygear-server is not compatible with 1.13.2 forget_password plugins;
1. The scheme make it easy to implement version checking at skygear-server / plugins startups or when client SDK connects with server.

Major change is probably on the release management of the software.

# API Design

**Remove this section if the feature have no API**

## Scenario

**Put list of Scenario how developers will use this API**

# Open Questions

**Put a list of open questions here before a complete design / specification is decided**

# Related Issues

- Server Issues
- Client Issues
- Guides Issues

# Progress Tracker

- [ ] Specification Design Approval
- [ ] Write (code + tests + API docs) then get them merged. **Put All PR there**
  - [ ] Code
  - [ ] Minimal tests
  - [ ] Minimal API Docs
- [ ] Guides. **Put All PR there**
- [ ] Release
  - [ ] Update release notes
  - [ ] Release

# Advice

- Specification Design Approval
  - Once you get LGTM from another Skygear Core Team Member, you can check this checkbox. And apply the "workflow/design-complete" label.
- Coding
  - Use as many PRs as you need.  Write tests in the same or different PRs, as is convenient for you.
  - API doc should goes in the same PR with the code.
  - As each PR is merged, add a comment to this issue referencing the PRs.
  - When you are done with the code, apply the "workflow/code-complete" label.
- Guides
  - Write or modify guides and get them merged in https://github.com/skygeario/guides
  - When the PR of guides is merged, check this checkbox and apply the "workflow/guides-complete" label.

**Comment by carmenlau** [comment]
I discussed with @rickmak before. I think we had this already. Only forgot password missed the tag :P 

In core, we used the minor version to identify the compatibility. So when server have a new minor version release, all the core SDK will also have the same minor release. Even there are no change for the SDK, we will still bump the minor version to match the server.

For plugin, if there is no change, we will not do any release. When release is needed for plugin, the plugin version will jump to the current server minor version.

We are going to release new version forgot password plugin (new reset password and user verification support), and we should follow the rule to do the release!
